services/streammanager/kafka/kafkamanager.go

```diff
@@ -83,268 +153,243 @@ func loadConfig() {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
 
-	if isSslEnabled {
-		caCertificate := destConfig.CACertificate
-		conf.Net.TLS.Enable = true
-		if caCertificate != "" {
-			tlsConfig := newTLSConfig(caCertificate)
-			if tlsConfig != nil {
-				conf.Net.TLS.Config = tlsConfig
+	clientConf := client.Config{
+		DialTimeout: kafkaDialTimeout,
+	}
+	if destConfig.SslEnabled {
+		if destConfig.CACertificate != "" {
+			clientConf.TLS = &client.TLS{
+				CACertificate: []byte(destConfig.CACertificate),
+				Cert:          clientCert,
+				Key:           clientKey,
 			}
+		} else {
+			clientConf.TLS = &client.TLS{WithSystemCertPool: true}
 		}
-		if destConfig.UseSASL {
-			// SASL is enabled only with SSL
-			err = setSASLConfig(conf, destConfig)
+
+		if destConfig.UseSASL { // SASL is enabled only with SSL
+			clientConf.SASL = &client.SASL{
+				Username: destConfig.Username,
+				Password: destConfig.Password,
+			}
+			clientConf.SASL.ScramHashGen, err = client.ScramHashGeneratorFromString(destConfig.SaslType)
 			if err != nil {
-				return nil, fmt.Errorf("[Kafka] Error while setting SASL config :: %w", err)
+				return nil, fmt.Errorf("[Kafka] invalid SASL type: %w", err)
 			}
 		}
 	}
 
-	return sarama.NewSyncProducer(hosts, conf)
-}
-
-// Sets SASL authentication config for Kafka
-func setSASLConfig(config *sarama.Config, destConfig Config) (err error) {
-	config.Net.SASL.Enable = true
-	config.Net.SASL.User = destConfig.Username
-	config.Net.SASL.Password = destConfig.Password
-	config.Net.SASL.Handshake = true
-	switch destConfig.SaslType {
-	case "plain":
-		config.Net.SASL.Mechanism = sarama.SASLTypePlaintext
-	case "sha512":
-		config.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient {
-			return &xdgSCRAMClient{HashGeneratorFcn: SHA512}
+	var c *client.Client
+	// The only component that supports multiple hosts is the consumer which we are not using here.
+	// It is also worth mentioning that internally the list of brokers should be populated accordingly anyway given
+	// that Kafka uses the "advertised.listeners" property to let clients know where they can connect.
+	// see: https://kafka.apache.org/documentation/#brokerconfigs_advertised.listeners
+	for _, host := range strings.Split(destConfig.HostName, ",") {
+		c, err = client.New("tcp", host+":"+destConfig.Port, clientConf)
+		if err != nil {
+			continue
 		}
-		config.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
-	case "sha256":
-		config.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient {
-			return &xdgSCRAMClient{HashGeneratorFcn: SHA256}
+		ctx, cancel := context.WithTimeout(context.TODO(), kafkaDialTimeout)
+		err = c.Ping(ctx)
+		cancel()
+		if err != nil {
+			continue
 		}
-		config.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA256
-	default:
-		return fmt.Errorf("[Kafka] invalid SASL type %s", destConfig.SaslType)
+
+		p, err := c.NewProducer(destConfig.Topic, client.ProducerConfig{
+			WriteTimeout: kafkaWriteTimeout,
```

Comment:
in the [previous implementation](https://github.com/rudderlabs/rudder-server/blob/master/services/streammanager/kafka/kafkamanager.go#L102) we were using `kafkaWriteTimeout` for setting the client's `ReadTimeout` as well. Is this omitted intentionally in the new implementation?

Reply:
Yes, I'm thus applying the default. If we want it I can add a real setting for it in the kafkamanager.

Reply:
The current readTimeout is 2s, whereas we will now be using 10s. If this extra delay of 8s is worthy and won't cause unwanted delays in the stream's processing, then it's fine :)

Reply:
I'm not sure 2s makes a lot of sense, I'll ask around about the rationale behind it just to be sure, but it feels like 10s is more appropriate especially since some Kafkas might not be on premise.

Reply:
`kafkaReadTimeout` introduced, if we want we can change it accordingly via configuration now.